src/github/github.controller.ts

```diff
@@ -0,0 +1,225 @@
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Put(':repoName/practices')
+  async updatePractices(
+    @Req() req: TRequest,
+    @Param('repoName') repoName: string,
+    @Body() updatePracticesDto: UpdatePracticesDto,
+  ) {
+    try {
+      return await this.githubService.updatePractices(
+        req,
+        repoName,
+        updatePracticesDto,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Get(':repoName')
+  async repoDetails(@Req() req: TRequest, @Param('repoName') repoName: string) {
+    try {
+      return await this.githubService.repoDetails(req, repoName);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Get('user/organizations')
+  async userOrganizations(@Req() req: TRequest) {
+    try {
+      return await this.githubService.userOrganizations(req);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Delete(':repoName/practices')
+  async deletePractices(
+    @Req() req: TRequest,
+    @Param('repoName') repoName: string,
+  ) {
+    try {
+      return await this.githubService.deletePractices(req, repoName);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Get(':repoName/pulls')
+  async getPullRequests(
+    @Req() req: TRequest,
+    @Param('repoName') repoName: string,
+    @Query('page') page: number = 1,
+    @Query('limit') limit: number = 20,
+  ) {
+    try {
+      return await this.githubService.getPullRequests(
+        req,
+        repoName,
+        page,
+        limit,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('user-details/sample/data')
+  async getUserDetails(@Query('accessToken') accessToken: string) {
+    const response = await this.githubService.getUserDetails(accessToken);
+    console.log(response);
+  }
```

Comment:
PullProofAI

The `getUserDetails` method lacks a return statement, which may lead to unexpected behavior. Ensure to return the response from the service call.

```diff
+     // Add a return statement to return the response from the service call
```


<!-- This is an auto-generated comment by PullProofAI -->